include/openssl/bio.h.in

```diff
@@ -520,7 +520,7 @@ int BIO_read_filename(BIO *b, const char *name);
 # define BIO_dup_state(b,ret)    BIO_ctrl(b,BIO_CTRL_DUP,0,(char *)(ret))
 
 # define BIO_reset(b)            (int)BIO_ctrl(b,BIO_CTRL_RESET,0,NULL)
-# define BIO_eof(b)              (int)BIO_ctrl(b,BIO_CTRL_EOF,0,NULL)
+# define BIO_eof(b)              (BIO_ctrl(b,BIO_CTRL_EOF,0,NULL) == 1)
```

Comment:
IMO this is incorrect. It should be `BIO_ctrl(b,BIO_CTRL_EOF,0,NULL) != 0` instead. It is much more safe to assume error means EOF than the other way around as an error means that you most probably won't be able to read any data anyway. Also if the current callers just used if (BIO_eof(b)) your proposal will change the behavior.

Reply:
Good point - done.
This is also more consistent with existing use of the eof() functions.